pkg/ebpf/telemetry/errors_telemetry_test.go

```diff
@@ -0,0 +1,193 @@
+	require.NoError(t, err)
+	m.Start()
+
+	_, err = os.Open("/proc/self/exe")
+	require.NoError(t, err)
+
+	ch := make(chan prometheus.Metric)
+	go func() {
+		collector.Collect(ch)
+		close(ch)
+	}()
+
+	//collect metrics from the channel
+	var metrics []prometheus.Metric
+	for m := range ch {
+		metrics = append(metrics, m)
+	}
+
+	return metrics
+}
+
+func TestMapsTelemetry(t *testing.T) {
+	mapsTelemetry := triggerTestAndGetTelemetry(t)
+	t.Cleanup(func() {
+		m.Stop(manager.CleanAll)
+	})
+
+	errorMapEntryFound, e2bigErrorFound := false, false
+	for _, promMetric := range mapsTelemetry {
+		dtoMetric := dto.Metric{}
+		assert.NoError(t, promMetric.Write(&dtoMetric), "Failed to parse metric %v", promMetric.Desc())
+		assert.NotNilf(t, dtoMetric.GetCounter(), "expected metric %v to be of a counter type", promMetric.Desc())
+
+		for _, label := range dtoMetric.GetLabel() {
+			switch label.GetName() {
+			case "map_name":
+				if label.GetValue() == "error_map" {
+					errorMapEntryFound = true
+				}
+			case "error":
+				if label.GetValue() == "E2BIG" {
+					e2bigErrorFound = true
+				}
+			}
+		}
+
+		// check error value immediately if map is discovered
+		if errorMapEntryFound {
+			require.True(t, e2bigErrorFound)
+		}
+	}
+
+	// ensure test fails if map telemetry not found
+	require.True(t, errorMapEntryFound)
+}
+
+func TestMapsTelemetrySuppressError(t *testing.T) {
+	mapsTelemetry := triggerTestAndGetTelemetry(t)
+	t.Cleanup(func() {
+		m.Stop(manager.CleanAll)
+	})
+
+	suppressMapEntryFound := false
+	for _, promMetric := range mapsTelemetry {
+		dtoMetric := dto.Metric{}
+		assert.NoError(t, promMetric.Write(&dtoMetric), "Failed to parse metric %v", promMetric.Desc())
+		assert.NotNilf(t, dtoMetric.GetCounter(), "expected metric %v to be of a counter type", promMetric.Desc())
+
+		for _, label := range dtoMetric.GetLabel() {
+			switch label.GetName() {
+			case "map_name":
+				if label.GetValue() == "suppress_map" {
+					suppressMapEntryFound = true
+				}
+			}
+		}
+
+		require.False(t, suppressMapEntryFound)
+	}
+}
+
+func TestHelpersTelemetry(t *testing.T) {
+	helperTelemetry := triggerTestAndGetTelemetry(t)
+	t.Cleanup(func() {
+		m.Stop(manager.CleanAll)
+	})
+
+	probeReadHelperFound, efaultErrorFound := false, false
+	for _, promMetric := range helperTelemetry {
+		dtoMetric := dto.Metric{}
+		assert.NoError(t, promMetric.Write(&dtoMetric), "Failed to parse metric %v", promMetric.Desc())
+		assert.NotNilf(t, dtoMetric.GetCounter(), "expected metric %v to be of a counter type", promMetric.Desc())
+
+		for _, label := range dtoMetric.GetLabel() {
+			switch label.GetName() {
+			case "helper":
+				if label.GetValue() == "bpf_probe_read" {
+					probeReadHelperFound = true
+				}
+			case "error":
+				if label.GetValue() == "EFAULT" {
+					efaultErrorFound = true
+				}
+			}
+
+			// make sure bpf_probe_read helper has an associated EFAULT error
+			if probeReadHelperFound {
+				require.True(t, efaultErrorFound)
```

Comment:
this could be after `require.True(t, probeReadHelperFound)`?

Reply:
same comment as above